### PR TITLE
Add support for Zstandard content-type

### PR DESCRIPTION
**Pull request by dlindenkreuz**
Although Req has long supported Zstandard `content-encoding`, it does not yet support Zstandard `content-type` which is used for original `.zst` archive files (as opposed to encoding other original media types like JSON for transfer).

This PR adds automatic decompression when `:ezstd` is available and `content-type: application/zstd` is detected. It depends on an upstream change in the MIME package: https://github.com/elixir-plug/mime/pull/82

### CODE REVIEW

**Comment by wojtekmach** [comment]
Thanks for the PR, please add docs and test and this is ready to go!

File: lib/req/steps.ex [hunk 1]
```diff
@@ -1569,6 +1569,20 @@ defmodule Req.Steps do
...
   end
 
+  defp decode_body({request, response}, "zst") do
+    if ezstd_loaded?() do
+      case :ezstd.decompress(response.body) do
+        decompressed when is_binary(decompressed) ->
+          {request, put_in(response.body, decompressed)}
+
+        {:error, _} ->
+          {request, %Req.ArchiveError{format: :zstd, data: response.body}}
```
Comment (wojtekmach) [inline lib/req/steps.ex:1579]: zstd is not an archive like tar and zip though, I don't think ArchiveError is appropriate here. please return a RuntimeError with custom message.

**Comment by dlindenkreuz** [comment]
Done. In the meanwhile, MIME has been updated to 2.0.6, which adds support for Zstd.

Req currently requires MIME `~> 1.6 or ~> 2.0` — should we leave a note somewhere that MIME `~> 2.0.6` is required for Zstd?

File: test/req/steps_test.exs [hunk 2]
```diff
@@ -870,6 +870,40 @@ defmodule Req.StepsTest do
...
+
+    test "zstd invalid" do
+      plug = fn conn ->
+        conn
+        |> Plug.Conn.put_resp_content_type("application/zstd", nil)
+        |> Plug.Conn.send_resp(200, "bad")
+      end
+
+      assert {:error, e} = Req.get(plug: plug)
+      assert match?(%RuntimeError{}, e)
```
Comment (wojtekmach) [inline test/req/steps_test.exs:901]:
```suggestion
      assert %RuntimeError{} = e
```

**Comment by wojtekmach** [comment]
Thank you!

**Pull request merged**